### ISSUE TITLE
SA1518 message misleading when newlineAtEndOfFile is set to require

### ISSUE THREAD

**Issue by NikolayIT** [DotNetAnalyzers/StyleCopAnalyzers#1847]
##### stylecop.json:

```json
{
  "settings": {
    "layoutRules": {
	  "newlineAtEndOfFile": "require"
    }
  }
}
```

##### Code file:
```cs
namespace SomeNamespace
{
    public class SomeClass
    {
    }
} // <--- No new line here
```

##### Analyzer warning:
    Warning	SA1518	Code must not contain blank lines at end of file

**In fact my code SHOULD contain blank line.**

P.S. thanks for implementing this setting :)

**Comment by sharwell** [comment]
I agree completely. I was able to make the code fix make sense:

> Fix whitespace at the end of the file

However, the diagnostic itself could still be improved. I was hoping to have a single message which applied regardless of the configuration or type of mistake (too few newlines, too many newlines). Do you have any suggestions?

:memo: Until we resolve this, I left the message in a form that makes sense for users with the *default* configuration, which also is the StyleCop Classic behavior.

**Comment by NikolayIT** [comment]
Can't we make different descriptors here: https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/95acc4eb10685286c8ce0785b500cde5dcd16b39/StyleCop.Analyzers/StyleCop.Analyzers/LayoutRules/SA1518CodeMustNotContainBlankLinesAtEndOfFile.cs and return them according to `settings.LayoutRules.NewlineAtEndOfFile`?

I can make a pull request for this if you wish.

The messages I am proposing are the ones from the [documentation](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/Configuration.md#lines-at-end-of-file):
* **allow** - Code must not contain blank lines at end of file
* **require** - Code is required to end with a single newline character
* **omit** - Code may not end with a newline character

**Comment by sharwell** [comment]
Those do look like good messages :blush: 

Go for it :)
